### ISSUE TITLE
Inconsistent behavior of sum().over() when summing up "null" values

### ISSUE THREAD

**Issue by denis-zahariev** [pola-rs/polars#6650]
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Issue description

The sum of ```null``` (i.e. ```None```) column values produced by calling ```pl.col(...).sum().over(...)``` is not consistent. Most of the time the result of the summation is ```null``` as expected, but occasionally the result is ```0.0```

### Reproducible example

```python
df = pl.DataFrame({
    "group": [1, 1, 2, 1, 1, 2,   1, 1, 2, 1, 1, 2],
    "value": [0, 0, 0, 0, 0, 0,   None, None, None, None, None, None],
    "time":  [1, 1, 1, 2, 2, 2,   3, 3, 3, 4, 4, 4], # does not work
    #"time":  [2, 2, 2, 2, 2, 2,   3, 3, 3, 4, 4, 4], # works
    #"time":  [1, 2, 2, 2, 2, 2,   3, 3, 3, 4, 4, 4], # works
    #"time":  [1, 1, 2, 2, 2, 2,   3, 3, 3, 4, 4, 4], # works
    #"time":  [1, 1, 1, 2, 2, 2,   3, 3, 3, 4, 4, 4], # does not work
    #"time":  [1, 1, 1, 1, 2, 2,   3, 3, 3, 4, 4, 4], # does not work (see above)
    #"time":  [1, 1, 1, 1, 1, 2,   3, 3, 3, 4, 4, 4], # works
    #"time":  [1, 1, 1, 1, 1, 1,   3, 3, 3, 4, 4, 4], # works
    #"time":  [1, 1, 1, 2, 2, 2,   3, 3, 3, 3, 3, 3], # works
    #"time":  [1, 1, 1, 2, 2, 2,   3, 3, 3, 3, 3, 4], # works
    #"time":  [1, 1, 1, 2, 2, 2,   3, 3, 3, 3, 4, 4], # does not work
    #"time":  [1, 1, 1, 2, 2, 2,   3, 3, 3, 4, 4, 4], # does not work (see above)
    #"time":  [1, 1, 1, 2, 2, 2,   3, 3, 4, 4, 4, 4], # works
    #"time":  [1, 1, 1, 2, 2, 2,   3, 4, 4, 4, 4, 4], # works
    #"time":  [1, 1, 1, 2, 2, 2,   4, 4, 4, 4, 4, 4], # works
})
RUNS = 1000
error_counter = 0
for i in range(RUNS):
    df = df.with_columns([
        pl.col("value").sum().over(["time", "group"]).alias("group_sum")
    ])
    # all sums for time>=3 should be null and not 0
    if not df.filter((pl.col("time") >= 3) & (pl.col("group_sum") == 0)).is_empty():
        error_counter += 1
print(f"{error_counter}/{RUNS} runs failed")
```

### Expected behavior

The result of the summation should have a consistent value in every run. 

### Installed versions

<details>

```
---Version info---
Polars: 0.16.2
Index type: UInt32
Platform: Windows-10-10.0.19045-SP0
Python: 3.10.7 (tags/v3.10.7:6cc6b13, Sep  5 2022, 14:08:36) [MSC v.1933 64 bit (AMD64)]
---Optional dependencies---
pyarrow: 11.0.0
pandas: 1.5.3
numpy: 1.24.1
fsspec: <not installed>
connectorx: <not installed>
xlsx2csv: 0.8.1
deltalake: <not installed>
matplotlib: <not installed>
```

</details>

**Comment by datenzauberai** [comment]
That's really weird. Maybe a race condition and missing initialization?

**Comment by ritchie46** [comment]
I cannot reproduce this locally? Somebody else seeing this?

**Comment by zundertj** [comment]
Cannot reproduce either (Linux Mint). Someone with a Windows install?

**Comment by datenzauberai** [comment]
I can reproduce this on Windows, but not on MacOS?!

**Comment by cmdlineluser** [comment]
Reproduced on MacOS.
```
29/1000 runs failed
```

**Comment by datenzauberai** [comment]
Interesting: it won't fail with a simple groupby/agg.
@ritchie46 Any hint how we can help to track this down to the root cause?

**Comment by cmdlineluser** [comment]
Everything seems to be the same up until:

https://github.com/pola-rs/polars/blob/master/polars/polars-lazy/src/physical_plan/expressions/window.rs#L563

`set_by_groups` calls `set_numeric` and it ends up entering this block:

https://github.com/pola-rs/polars/blob/master/polars/polars-lazy/src/physical_plan/expressions/window.rs#762

When I add a `dbg!()` inside that block at line 763 I can no longer reproduce the bug...

**Comment by cmdlineluser** [comment]
I managed to reproduce the error with a println for `idx` and `v` inside this block:

https://github.com/pola-rs/polars/blob/master/polars/polars-lazy/src/physical_plan/expressions/window.rs#L775

The values are the same in both the error and non-error cases.

Maybe this suggests the difference happens in the `None` case?

The error case happens way less often with the `println` added.

I tried adding println inside the `None` case but have not been able to reproduce the error as of yet - even with `200,000` iterations.
